### PR TITLE
DATAUP-532: housekeeping

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

A drop of housekeeping before working on DATAUP-532. This comprises:

- renaming the `jobStateList` widget to the more appropriate `jobStatusTable`
- moving the `jobCommChannel` widget into the `common` directory
- some message renaming:
- - `job-retried` message to `job-retry-response`
- - `request-job-completion` to `request-job-updates-stop`
- - `request-job-update` to `request-job-updates-start`
- removed `job_canceled` message type, which never gets sent by the backend

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-532
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and seeing that nothing has changed

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: nbextensions/appCell2/widgets/tabs/status/jobStateList.js [hunk 1]
```diff
@@ -1,51 +1,40 @@
...
+    UI,
+    html,
+    JobStateListRow
+) => {
     'use strict';
 
     const t = html.tag,
-        div = t('div'),
-        p = t('p'),
-        span = t('span'),
```
Comment (ialarmedalien) [inline nbextensions/appCell2/widgets/tabs/status/jobStateList.js:14]: unused

File: kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobActionDropdown.js [hunk 2]
```diff
@@ -189,7 +189,7 @@ define([
          * @param {object} args  -- with keys
          *      node:       DOM node to attach to
          *
-         * @returns {Promise} started JobStateList widget
+         * @returns {Promise} started JobActionDropdown widget
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobActionDropdown.js:192]: ooops!

File: kbase-extension/static/kbase/js/kbaseNarrative.js [hunk 3]
```diff
@@ -1207,8 +1205,7 @@ define([
                 allSpecs[spec.info.id] = spec;
                 return allSpecs;
             }, {});
-            const cell = this.insertAndSelectCellBelow('code', null, cellData);
-            return cell;
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/kbaseNarrative.js:1211]: no need for `cell`

File: kbase-extension/static/narrative_paths.js [hunk 4]
```diff
@@ -121,7 +121,7 @@ require.config({
         kbaseNarrativeDataPanel: 'kbase/js/widgets/narrative_core/kbaseNarrativeDataPanel',
         kbaseNarrativeDataList: 'kbase/js/widgets/narrative_core/kbaseNarrativeDataList',
         kbaseNarrativeSidePanel: 'kbase/js/widgets/narrative_core/kbaseNarrativeSidePanel',
-        jobCommChannel: 'kbase/js/widgets/narrative_core/jobCommChannel',
```
Comment (ialarmedalien) [inline kbase-extension/static/narrative_paths.js:124]: move to `common/` dir

File: nbextensions/appCell2/widgets/tabs/status/jobStateList.js [hunk 5]
```diff
@@ -1,51 +1,40 @@
-define([
-    'bluebird',
-    'common/runtime',
-    'common/ui',
-    'common/format',
```
Comment (ialarmedalien) [inline nbextensions/appCell2/widgets/tabs/status/jobStateList.js:5]: unused import

File: nbextensions/appCell2/widgets/tabs/status/jobStateList.js [hunk 6]
```diff
@@ -1,51 +1,40 @@
...
             widgets = {},
-            model = config.model,
-            parentJobId,
-            parentListener;
+            model = config.model;
+        let container, parentJobId, parentListener;
 
         function createTableRow(id) {
-            const table = container.getElementsByTagName('tbody')[0];
+            const tableEl = container.getElementsByTagName('tbody')[0];
```
Comment (ialarmedalien) [inline nbextensions/appCell2/widgets/tabs/status/jobStateList.js:25]: name used in outer scope

File: nbextensions/appCell2/widgets/tabs/status/jobStateList.js [hunk 7]
```diff
@@ -119,15 +108,15 @@ define([
                 jobId: jobId,
                 initialState: initialState,
                 isParentJob: isParentJob ? true : false,
-                clickFunction: function (jobRow, jobId, isParentJob) {
+                clickFunction: function (jobRow, _jobId, _isParentJob) {
```
Comment (ialarmedalien) [inline nbextensions/appCell2/widgets/tabs/status/jobStateList.js:111]: names already used in outer scope

File: nbextensions/appCell2/widgets/tabs/status/logTab.js [hunk 8]
```diff
@@ -7,10 +7,9 @@ define([
     'common/ui',
     'common/runtime',
     'util/jobLogViewer',
-    'util/jobStateViewer',
     './jobStateList',
     './jobInputParams',
-], (Promise, html, UI, Runtime, JobLogViewer, JobStateViewer, JobStateList, JobInputParams) => {
```
Comment (ialarmedalien) [inline nbextensions/appCell2/widgets/tabs/status/logTab.js:13]: jobStateViewer unused

File: kbase-extension/static/kbase/js/common/jobCommChannel.js [hunk 9]
```diff
@@ -270,13 +270,6 @@ define([
                     break;
 
                 // JOB messages
-                case 'job_canceled':
-                    this.sendBusMessage(JOB, msgData.job_id, 'job-canceled', {
-                        jobId: msgData.job_id,
-                        via: 'job_canceled',
-                    });
-                    break;
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobCommChannel.js:278]: There is no message of type `job-canceled` that gets sent by the backend

File: test/unit/spec/common/jobCommChannel-spec.js [hunk 10]
```diff
@@ -403,28 +403,6 @@ define(['jobCommChannel', 'base/js/namespace', 'common/runtime', 'testUtil'], (
             });
         });
 
-        it('Should send job_canceled message to the bus', () => {
```
Comment (ialarmedalien) [inline test/unit/spec/common/jobCommChannel-spec.js:406]: this message is never sent by the backend

File: kbase-extension/static/kbase/js/common/jobCommChannel.js (hunk 9 above)
Comment (briehl) [reply kbase-extension/static/kbase/js/common/jobCommChannel.js:278]: 🤦

File: kbase-extension/static/kbase/js/kbaseNarrative.js [hunk 11]
```diff
@@ -13,7 +13,7 @@ define([
     'bluebird',
     'handlebars',
     'narrativeConfig',
-    'jobCommChannel',
```
Comment (briehl) [inline kbase-extension/static/kbase/js/kbaseNarrative.js:16]: You can still use `jobCommChannel` here as you're changing the alias in narrative_paths.js. Or just remove the alias from there and go by the `common` directory instead.

File: test/unit/spec/common/jobCommChannel-spec.js [hunk 12]
```diff
@@ -1,4 +1,4 @@
-define(['jobCommChannel', 'base/js/namespace', 'common/runtime', 'testUtil'], (
```
Comment (briehl) [inline test/unit/spec/common/jobCommChannel-spec.js:1]: As before, you don't need to set `common/jobCommChannel` as long as the alias in `narrative_paths.js` is updated.

**Review by briehl — CHANGES_REQUESTED** [review]
A couple repeated suggestions, but looks good.

Honestly, since `jobCommChannel` is really only used (and only intended to be used) by `kbaseNarrative.js`, it doesn't need to have an alias in `narrative_paths.js`, and we can just set it as a dependency with `common/jobCommChannel`.

**Review by briehl — APPROVED** [review]

**Pull request merged**